File: frontend-gym/src/lib/db/models/progreso.js
const { DataTypes } = require('sequelize');
const sequelize = require('../sequelize');

const Progreso = sequelize.define('Progreso', {
  id_progreso: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  id_usuario: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'usuarios',
      key: 'id_usuario'
    }
  },
  fecha: {
    type: DataTypes.DATEONLY,
    allowNull: false,
    defaultValue: DataTypes.NOW
  },
  peso: {
    type: DataTypes.DECIMAL(5, 2),
    allowNull: true
  },
  porcentaje_grasa: {
    type: DataTypes.DECIMAL(5, 2),
    allowNull: true
  },
  pecho: {
    type: DataTypes.DECIMAL(5, 2),
    allowNull: true
  },
  cintura: {
    type: DataTypes.DECIMAL(5, 2),
    allowNull: true
  },
  brazo: {
    type: DataTypes.DECIMAL(5, 2),
    allowNull: true
  },
  pierna: {
    type: DataTypes.DECIMAL(5, 2),
    allowNull: true
  },
  cadera: {
    type: DataTypes.DECIMAL(5, 2),
    allowNull: true
  },
  observaciones: {
    type: DataTypes.TEXT,
    allowNull: true
  }
}, {
  tableName: 'progresos',
  timestamps: false,
});

module.exports = Progreso;

File: arenagymapk/lib/models/progreso.dart
class Progreso {
  final int idProgreso;
  final int idUsuario;
  final String fecha;
  final double? peso;
  final double? porcentajeGrasa;
  final double? pecho;
  final double? cintura;
  final double? brazo;
  final double? pierna;
  final double? cadera;
  final String? observaciones;

  Progreso({
    required this.idProgreso,
    required this.idUsuario,
    required this.fecha,
    required this.peso,
    required this.porcentajeGrasa,
    required this.pecho,
    required this.cintura,
    required this.brazo,
    required this.pierna,
    required this.cadera,
    required this.observaciones,
  });

  factory Progreso.fromJson(Map<String, dynamic> json) {
    return Progreso(
      idProgreso: json['id_progreso'] as int,
      idUsuario: json['id_usuario'] as int? ?? 0,
      fecha: json['fecha']?.toString() ?? '',
      peso: double.tryParse(json['peso']?.toString() ?? ''),
      porcentajeGrasa: double.tryParse(json['porcentaje_grasa']?.toString() ?? ''),
      pecho: double.tryParse(json['pecho']?.toString() ?? ''),
      cintura: double.tryParse(json['cintura']?.toString() ?? ''),
      brazo: double.tryParse(json['brazo']?.toString() ?? ''),
      pierna: double.tryParse(json['pierna']?.toString() ?? ''),
      cadera: double.tryParse(json['cadera']?.toString() ?? ''),
      observaciones: json['observaciones'] as String?,
    );
  }
}
